system-addon/content-src/components/TopSites/TopSitesEdit.jsx

```diff
@@ -71,14 +71,29 @@ class TopSitesEdit extends React.PureComponent {
   }
   render() {
     const realTopSites = this.props.TopSites.rows.slice(0, this.props.TopSitesCount);
-    const placeholderCount = this.props.TopSitesCount - realTopSites.length;
     const showEditForm = (this.props.TopSites.editForm && this.props.TopSites.editForm.visible) ||
                          (this.state.showEditModal && this.state.showEditForm);
     let editIndex = this.state.editIndex;
     if (showEditForm && this.props.TopSites.editForm.visible) {
       const targetURL = this.props.TopSites.editForm.site.url;
       editIndex = this.props.TopSites.rows.findIndex(s => s.url === targetURL);
     }
+    const topSitesUI = [];
+    for (let i = 0, l = this.props.TopSitesCount; i < l; i++) {
+      const link = realTopSites[i];
+      if (link) {
+        topSitesUI.push(<TopSite
+          key={link.guid || link.url}
+          dispatch={this.props.dispatch}
+          link={link}
+          index={i}
+          intl={this.props.intl}
+          onEdit={this.onEdit}
+          editMode={true} />);
+      } else {
+        topSitesUI.push(<TopSitePlaceholder key={i} />);
+      }
+    }
```

Comment:
Thinking about how to DRY this up a little. I think a new `TopSiteList` component that just does this should work. *tries*